algebra/src/groups/mod.rs

```diff
@@ -1,83 +1,60 @@
-use crate::UniformRand;
 use crate::{
-    BitIterator, CanonicalDeserialize, CanonicalSerialize, FromBytesChecked, SemanticallyValid,
+    bytes::{FromBytes, ToBytes},
+    fields::PrimeField,
 };
+use crate::{CanonicalDeserialize, CanonicalSerialize, FromBytesChecked, SemanticallyValid};
 use std::{
     fmt::{Debug, Display},
     hash::Hash,
-    ops::{Add, AddAssign, Neg, Sub, SubAssign},
+    ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub, SubAssign},
 };
+use num_traits::Zero;
 
-use crate::{
-    bytes::{FromBytes, ToBytes},
-    fields::PrimeField,
-};
-use serde::{Deserialize, Serialize};
+mod group_vec;
+pub use group_vec::*;
 
 #[cfg(test)]
 pub mod tests;
 
 pub trait Group:
-    ToBytes
+    'static
+    + ToBytes
     + FromBytes
     + FromBytesChecked
     + SemanticallyValid
-    + Serialize
-    + for<'a> Deserialize<'a>
     + CanonicalSerialize
     + CanonicalDeserialize
-    + Copy
     + Clone
     + Debug
     + Display
     + Default
     + Send
     + Sync
-    + 'static
     + Eq
     + Hash
+    + Zero
     + Neg<Output = Self>
-    + UniformRand
+    + Add<Self, Output = Self>
+    + Sub<Self, Output = Self>
+    + AddAssign<Self>
+    + SubAssign<Self>
     + for<'a> Add<&'a Self, Output = Self>
     + for<'a> Sub<&'a Self, Output = Self>
+    + for<'a> Mul<&'a <Self as Group>::ScalarField, Output = Self>
     + for<'a> AddAssign<&'a Self>
     + for<'a> SubAssign<&'a Self>
+    + for<'a> MulAssign<&'a <Self as Group>::ScalarField>
 {
-    type ScalarField: PrimeField + Into<<Self::ScalarField as PrimeField>::BigInt>;
-
-    /// Returns the additive identity.
-    fn zero() -> Self;
-
-    /// Returns `self == zero`.
-    fn is_zero(&self) -> bool;
+    type ScalarField: PrimeField;
```

Comment:
As pointed out in my review of [Poly commit refactoring #35](https://github.com/HorizenLabs/poly-commit/pull/35) I would opt to remove the scalar field from the group trait, which was rather a quick work-around needed for our polynomial commitment implementation.